storage/invoke.go

```diff
@@ -67,22 +68,40 @@ func run(ctx context.Context, call func(ctx context.Context) error, retry *retry
 		errorFunc = retry.shouldRetry
 	}
 
+	var quitAfterTimer *time.Timer
+	if retry.maxDuration != 0 {
+		quitAfterTimer = time.NewTimer(retry.maxDuration)
+		defer quitAfterTimer.Stop()
+	}
+
+	var lastErr error
 	return internal.Retry(ctx, bo, func() (stop bool, err error) {
+		if retry.maxDuration != 0 {
+			select {
+			case <-quitAfterTimer.C:
+				if lastErr == nil {
+					return true, errors.New("storage: request not sent, choose a larger value for the retry deadline")
+				}
+				return true, fmt.Errorf("storage: retry timed out after %v attempts; last error: %w", attempts, lastErr)
```

Comment:
This err should include the value of the timeout as well.

Reply:
Done